Bugfix IllegalArgumentException when ignoring telegram

shaguarger opened the pull request:
This bug kills the complete thread and no more telegrams will be received.

shaguarger committed:
Bugfix IllegalArgumentException when ignoring telegram

This bug kills the complete thread and no more telegrams will be received.

Signed-off-by: Arne Wohlert <16501863+shaguarger@users.noreply.github.com>

shaguarger committed:
General code improvements and correct calculation of humidity as defined in the vallox documentation

Signed-off-by: Arne Wohlert <16501863+shaguarger@users.noreply.github.com>

gitMiguel commented:
Hi @shaguarger ,

I encountered this bug also.

Regarding your other commit I'd like to know why you moved the checksum calculator in its own class? I mean there's nothing wrong about it but is there something in functionality that I'm not aware of?

/Miika

shaguarger commented:
I fixed this issue for a friend. I didn't see you PR - that would have saved me some time for bugfixing.

I extracted the checksum calculator for further unit testing in our next vallox binding session.

Btw: it would be nice to see my changes (updateCO2(..)) as at least Co-authored-by...

https://help.github.com/articles/creating-a-commit-with-multiple-authors/
